### ISSUE TITLE
vim-rails commands not recognised by vim 

### ISSUE THREAD

**Issue by AbhimanyuAryan** [tpope/vim-rails#446]
I followed [this guide][1] to install vim-rails. However when I am in controller/ and editing application_controller.rb, I pressed `esc` key and typed `:Rserver` and vim returned me `Not an editor command Rserver`. What is it that I am doing wrong? 

<b>Here is my vimrc file</b>

    set nocompatible              " be iMproved, required
    filetype off                  " required
    
    " set the runtime path to include Vundle and initialize
    set rtp+=~/.vim/bundle/Vundle.vim
    call vundle#begin()
    " alternatively, pass a path where Vundle should install plugins
    " call vundle#begin('~/some/path/here')
    
    " let Vundle manage Vundle, required
    Plugin 'gmarik/Vundle.vim'
    Plugin 'vim-scripts/ScrollColors'
    Plugin 'tpope/vim-rails'
    Plugin 'tpope/vim-bundler'
    Plugin 'edkolev/tmuxline.vim'
    Plugin 'kien/ctrlp.vim' " fuzzy find files
    Plugin 'scrooloose/nerdtree' " file drawer, open with :NERDTreeToggle
    Plugin 'benmills/vimux'
    Plugin 'tpope/vim-fugitive' " the ultimate git helper
    Plugin 'tpope/vim-commentary' " comment/uncomment lines with gcc or gc i
    
    call vundle#end()
    filetype plugin indent on
    
    
    
    
    set nocompatible                " choose no compatibility with legacy vi
    syntax enable
    set encoding=utf-8
    set showcmd                     " display incomplete commands
    filetype plugin indent on       " load file type plugins + indentation
    
    "" Whitespace
    set nowrap                      " don't wrap lines
    set tabstop=2 shiftwidth=2      " a tab is two spaces (or set this to 4)
    set expandtab                   " use spaces, not tabs (optional)
    set backspace=indent,eol,start  " backspace through everything in insert mode
    
    "" Searching
    set hlsearch                    " highlight matches
    set incsearch                   " incremental searching
    set ignorecase                  " searches are case insensitive...
    set smartcase                   " ... unless they contain at least one capital letter
    colorscheme developer
    
    
    """""""""""""""""""""""""""""""""""""""""""""""""""""""""""
    " => General Vim Configurations by Nick at Vim-Workshop
    """""""""""""""""""""""""""""""""""""""""""""""""""""""""""
    
    " load plugins from vundle
    filetype off
    
    set rtp+=~/.vim/bundle/vundle/
    call vundle#begin()
    
    " let vundle manage vundle
    Plugin 'gmarik/vundle'
    
    " utilities
    Plugin 'kien/ctrlp.vim' " fuzzy find files
    Plugin 'scrooloose/nerdtree' " file drawer, open with :NERDTreeToggle
    Plugin 'benmills/vimux'
    Plugin 'tpope/vim-fugitive' " the ultimate git helper
    Plugin 'tpope/vim-commentary' " comment/uncomment lines with gcc or gc in visual mode
    
    " colorschemes
    Plugin 'chriskempson/base16-vim'
    
    " JavaScript plugins
    Plugin 'pangloss/vim-javascript'
    Plugin 'jelera/vim-javascript-syntax'
    
    call vundle#end()
    filetype plugin indent on
    
    set nocompatible " not compatible with vi
    set autoread " detect when a file is changed
    
    " make backspace behave in a sane manner
    set backspace=indent,eol,start
    
    " set a map leader for more key combos
    let mapleader = ','
    
    " Tab control
    set noexpandtab " tabs ftw
    set smarttab " tab respects 'tabstop', 'shiftwidth', and 'softtabstop'
    set tabstop=4 " the visible width of tabs
    set softtabstop=4 " edit as if the tabs are 4 characters wide
    set shiftwidth=4 " number of spaces to use for indent and unindent
    set shiftround " round indent to a multiple of 'shiftwidth'
    
    set clipboard=unnamed
    
    " faster redrawing
    set ttyfast
    
    " code folding settings
    set foldmethod=syntax " fold based on indent
    set foldnestmax=10 " deepest fold is 10 levels
    set nofoldenable " don't fold by default
    set foldlevel=1
    
    """""""""""""""""""""""""""""""""""""""""""""""""""""""""""
    " => User Interface
    """""""""""""""""""""""""""""""""""""""""""""""""""""""""""
    
    " Searching
    set ignorecase " case insensitive searching
    set smartcase " case-sensitive if expresson contains a capital letter
    set hlsearch
    set incsearch " set incremental search, like modern browsers
    set nolazyredraw " don't redraw while executing macros
    
    set magic " Set magic on, for regex
    
    set showmatch " show matching braces
    set mat=2 " how many tenths of a second to blink
    
    " switch syntax highlighting on
    syntax on
    
    set encoding=utf8
    let base16colorspace=256  " Access colors present in 256 colorspace"
    set t_Co=256 " Explicitly tell vim that the terminal supports 256 colors"
    set background=dark
    colorscheme delek
    
    set number
    
    set autoindent " automatically set indent of new line
    set smartindent
    
    set laststatus=2 " show the satus line all the time
    
    """""""""""""""""""""""""""""""""""""""""""""""""""""""""""
    " => Mappings
    """""""""""""""""""""""""""""""""""""""""""""""""""""""""""
    
    map <leader>ev :e! ~/.vimrc<cr> " edit ~/.vimrc
    
    map <leader>wc :wincmd q<cr>
    
    " moving up and down work as you would expect
    nnoremap <silent> j gj
    nnoremap <silent> k gk
    
    " helpers for dealing with other people's code
    nmap \t :set ts=4 sts=4 sw=4 noet<cr>
    nmap \s :set ts=4 sts=4 sw=4 et<cr>
    
    """""""""""""""""""""""""""""""""""""""""""""""""""""""""""
    " => Functions
    """""""""""""""""""""""""""""""""""""""""""""""""""""""""""
    
    map <C-h> :call WinMove('h')<cr>
    map <C-j> :call WinMove('j')<cr>
    map <C-k> :call WinMove('k')<cr>
    map <C-l> :call WinMove('l')<cr>
    
    " Window movement shortcuts
    " move to the window in the direction shown, or create a new window
    function! WinMove(key)
        let t:curwin = winnr()
        exec "wincmd ".a:key
        if (t:curwin == winnr())
            if (match(a:key,'[jk]'))
                wincmd v
            else
                wincmd s
            endif
            exec "wincmd ".a:key
        endif
    endfunction
    
    """""""""""""""""""""""""""""""""""""""""""""""""""""""""""
    " => Plugin settings
    """""""""""""""""""""""""""""""""""""""""""""""""""""""""""
    
    " close NERDTree after a file is opened
    let g:NERDTreeQuitOnOpen=0
    " show hidden files in NERDTree
    let NERDTreeShowHidden=1
    " Toggle NERDTree
    nmap <silent> <leader>k :NERDTreeToggle<cr>
    " expand to the path of the file in the current buffer
    nmap <silent> <leader>y :NERDTreeFind<cr>
    
    " map fuzzyfinder (CtrlP) plugin
    " nmap <silent> <leader>t :CtrlP<cr>
    nmap <silent> <leader>r :CtrlPBuffer<cr>
    let g:ctrlp_map='<leader>t'
    let g:ctrlp_dotfiles=1
    let g:ctrlp_working_path_mode = 'ra'
    
    " CtrlP ignore patterns
    let g:ctrlp_custom_ignore = {
                \ 'dir': '\.git$\|node_modules$\|\.hg$\|\.svn$',
                \ 'file': '\.exe$\|\.so$'
                \ }
    
    " search the nearest ancestor that contains .git, .hg, .svn
    let g:ctrlp_working_path_mode = 2

  [1]: http://www.sitepoint.com/effective-rails-development-vim/

**Comment by tpope** [comment]
Does the `:Rails` command exist? Does your app have both `app/` and `config/environment.rb`?

**Comment by anhari** [comment]
Not sure if this helps, but I only experience this issue when pulling up a new tab in vim.

**Comment by tpope** [comment]
Correct.

**Comment by robwold** [comment]
I know this issue is closed, but @anhari 's solution doesn't actually seem to be the same as that of @AbhimanyuAryan .
I came here after following the same tutorial and having the same issue. I've found that the commands mentioned there like `:Rmodel` don't seem to work, but the ones prefixed with an 'E' (`:Emodel`) do. Is this just outdated info from that article? AFAICT from running `help :Rails` it looks as if this should be aliased:
`
                                rails-:Emodel rails-:Rmodel
:Emodel [{name}]        Edit the specified or current model.
`

Although I am a vim noob so it's possible I'm mis-reading this?
